server/src/main/java/org/apache/druid/discovery/DruidNodeDiscoveryProvider.java

```diff
@@ -235,6 +235,28 @@ public void nodeViewInitialized()
           }
         }
       }
+
+      @Override
+      public void nodeViewInitializedTimedOut()
+      {
+        synchronized (lock) {
+          if (uninitializedNodeRoles == 0) {
+            log.error("Unexpected call of nodeViewInitialized()");
+            return;
+          }
+          uninitializedNodeRoles--;
+          if (uninitializedNodeRoles == 0) {
+            for (Listener listener : listeners) {
+              try {
+                listener.nodeViewInitialized();
+              }
+              catch (Exception ex) {
+                log.error(ex, "Listener[%s].nodeViewInitialized() threw exception. Ignored.", listener);
```

Comment:
Should mention `nodeViewInitializedTimedOut()`, once the call is updated.

Reply:
Done